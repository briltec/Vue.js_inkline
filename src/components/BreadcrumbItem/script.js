import AttributesProviderMixin from '@inkline/inkline/src/mixins/components/providers/AttributesProviderMixin';
import ClassesProviderMixin from '@inkline/inkline/src/mixins/components/providers/ClassesProviderMixin';

import ActivePropertyMixin from '@inkline/inkline/src/mixins/components/properties/ActivePropertyMixin';
import SizePropertyMixin from '@inkline/inkline/src/mixins/components/properties/SizePropertyMixin';

export default {
    name: 'IBreadcrumbItem',
    mixins: [
        AttributesProviderMixin,
        ClassesProviderMixin,

        ActivePropertyMixin,
        SizePropertyMixin
    ],
    computed: {
        tag() {
            return this.attributes.to ? 'router-link' : 'a';
        }
    }
};