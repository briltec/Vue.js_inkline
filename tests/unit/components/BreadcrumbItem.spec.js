import { mount } from '@vue/test-utils';
import BreadcrumbItem from 'inkline/components/BreadcrumbItem';

const RouterLink = {
    name: 'RouterLink',
    render() {}
};

describe('Components', () => {
    describe('BreadcrumbItem', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(BreadcrumbItem, {
                components: {
                    RouterLink
                }
            });
        });

        it('should be named correctly', () => {
            expect(BreadcrumbItem.name).toEqual('IBreadcrumbItem');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('tag()', () => {
                it('should be defined', () => {
                    expect(wrapper.vm.tag).toBeDefined();
                });

                it('should be <a> tag by default', () => {
                    expect(wrapper.vm.tag).toEqual('a');
                });

                it('should be <router-link> tag if "to" attribute is provided', () => {
                    wrapper = mount(BreadcrumbItem, {
                        components: {
                            RouterLink
                        },
                        propsData: {
                            to: true
                        }
                    });

                    expect(wrapper.vm.tag).toEqual('router-link');
                });
            });
        });
    });
});
