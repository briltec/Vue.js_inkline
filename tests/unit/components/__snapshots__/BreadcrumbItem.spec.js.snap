// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components BreadcrumbItem should render correctly 1`] = `<li class="breadcrumb-item"><a></a></li>`;
